[Hue/Jue] Runtime/API Exception, Internal Error 404

```
2016-06-24 02:00:13.837 [INFO ] [smarthome.event.ItemCommandEvent    ] - Item 'hue_15_color' received command 5
2016-06-24 02:00:13.867 [ERROR] [ome.core.thing.internal.ThingManager] - Exception occured while calling handler: java.lang.RuntimeException: nl.q42.jue.exceptions.ApiException: Internal error, 404
java.util.concurrent.ExecutionException: java.lang.RuntimeException: nl.q42.jue.exceptions.ApiException: Internal error, 404
    at java.util.concurrent.FutureTask.report(FutureTask.java:122)[:1.8.0_91]
    at java.util.concurrent.FutureTask.get(FutureTask.java:206)[:1.8.0_91]
    at org.eclipse.smarthome.core.common.SafeMethodCaller.callAsynchronous(SafeMethodCaller.java:179)[96:org.eclipse.smarthome.core:0.8.0.201605311814]
    at org.eclipse.smarthome.core.common.SafeMethodCaller.call(SafeMethodCaller.java:72)[96:org.eclipse.smarthome.core:0.8.0.201605311814]
    at org.eclipse.smarthome.core.common.SafeMethodCaller.call(SafeMethodCaller.java:56)[96:org.eclipse.smarthome.core:0.8.0.201605311814]
    at org.eclipse.smarthome.core.thing.internal.ThingManager.receiveCommand(ThingManager.java:368)[102:org.eclipse.smarthome.core.thing:0.8.0.201605311814]
    at org.eclipse.smarthome.core.items.events.AbstractItemEventSubscriber.receive(AbstractItemEventSubscriber.java:46)[96:org.eclipse.smarthome.core:0.8.0.201605311814]
    at org.eclipse.smarthome.core.internal.events.OSGiEventManager$1.call(OSGiEventManager.java:192)[96:org.eclipse.smarthome.core:0.8.0.201605311814]
    at org.eclipse.smarthome.core.internal.events.OSGiEventManager$1.call(OSGiEventManager.java:1)[96:org.eclipse.smarthome.core:0.8.0.201605311814]
    at org.eclipse.smarthome.core.common.SafeMethodCaller$CallableWrapper.call(SafeMethodCaller.java:170)[96:org.eclipse.smarthome.core:0.8.0.201605311814]
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)[:1.8.0_91]
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)[:1.8.0_91]
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)[:1.8.0_91]
    at java.lang.Thread.run(Thread.java:745)[:1.8.0_91]
Caused by: java.lang.RuntimeException: nl.q42.jue.exceptions.ApiException: Internal error, 404
    at org.eclipse.smarthome.binding.hue.handler.HueBridgeHandler.updateLightState(HueBridgeHandler.java:198)[195:org.eclipse.smarthome.binding.hue:0.8.0.201605311814]
    at org.eclipse.smarthome.binding.hue.handler.HueLightHandler.handleCommand(HueLightHandler.java:202)[195:org.eclipse.smarthome.binding.hue:0.8.0.201605311814]
    at org.eclipse.smarthome.core.thing.internal.ThingManager$2.call(ThingManager.java:371)[102:org.eclipse.smarthome.core.thing:0.8.0.201605311814]
    at org.eclipse.smarthome.core.thing.internal.ThingManager$2.call(ThingManager.java:1)[102:org.eclipse.smarthome.core.thing:0.8.0.201605311814]
    ... 5 more
Caused by: nl.q42.jue.exceptions.ApiException: Internal error, 404
    at nl.q42.jue.HueBridge.handleErrors(HueBridge.java:829)[195:org.eclipse.smarthome.binding.hue:0.8.0.201605311814]
    at nl.q42.jue.HueBridge.setLightState(HueBridge.java:234)[195:org.eclipse.smarthome.binding.hue:0.8.0.201605311814]
    at org.eclipse.smarthome.binding.hue.handler.HueBridgeHandler.updateLightState(HueBridgeHandler.java:193)[195:org.eclipse.smarthome.binding.hue:0.8.0.201605311814]
    ... 8 more
2016-06-24 02:00:13.869 [INFO ] [smarthome.event.ItemCommandEvent    ] - Item 'hue_16_color' received command 5
2016-06-24 02:00:14.877 [INFO ] [smarthome.event.ItemCommandEvent    ] - Item 'hue_17_color' received command 5
2016-06-24 02:00:14.890 [INFO ] [smarthome.event.ItemCommandEvent    ] - Item 'hue_18_color' received command 5
2016-06-24 02:00:14.903 [INFO ] [smarthome.event.ItemCommandEvent    ] - Item 'hue_19_color' received command 5
2016-06-24 02:00:14.918 [INFO ] [smarthome.event.ItemCommandEvent    ] - Item 'hue_20_color' received command 5
2016-06-24 02:00:14.931 [INFO ] [smarthome.event.ItemCommandEvent    ] - Item 'hue_21_color' received command 5
2016-06-24 02:00:14.946 [INFO ] [smarthome.event.ItemCommandEvent    ] - Item 'hue_22_color' received command 5
2016-06-24 02:00:14.948 [INFO ] [marthome.event.ItemStateChangedEvent] - hue_21_color changed from 331,21,9 to 331,21,5
2016-06-24 02:00:14.948 [INFO ] [marthome.event.ItemStateChangedEvent] - hue_22_color changed from 331,21,9 to 331,21,5
2016-06-24 02:00:14.949 [INFO ] [marthome.event.ItemStateChangedEvent] - hue_8_color changed from 354,87,9 to 354,87,5
2016-06-24 02:00:14.949 [INFO ] [marthome.event.ItemStateChangedEvent] - hue_10_color changed from 34,99,9 to 34,99,5
2016-06-24 02:00:14.950 [INFO ] [marthome.event.ItemStateChangedEvent] - hue_11_color changed from 96,100,9 to 96,100,5
2016-06-24 02:00:14.950 [INFO ] [marthome.event.ItemStateChangedEvent] - hue_12_color changed from 331,21,9 to 331,21,5
2016-06-24 02:00:14.950 [INFO ] [marthome.event.ItemStateChangedEvent] - hue_13_color changed from 34,99,9 to 34,99,5
2016-06-24 02:00:14.952 [INFO ] [marthome.event.ItemStateChangedEvent] - hue_14_color changed from 331,21,9 to 331,21,5
2016-06-24 02:00:14.953 [INFO ] [marthome.event.ItemStateChangedEvent] - hue_15_color changed from 34,99,9 to 34,99,5
2016-06-24 02:00:14.953 [INFO ] [marthome.event.ItemStateChangedEvent] - hue_18_color changed from 34,99,9 to 34,99,5
2016-06-24 02:00:15.042 [INFO ] [marthome.event.ItemStateChangedEvent] - hue_20_color changed from 331,21,9 to 331,21,5
```

The setup involved item type `Color` hue bulbs being affected via a rule to control their brightness via KNX `Dimmer` item when it receives a command.

hue.items:

```
Group gHue "Hue Group"
Color  hue_8_color "Hue Lamp 3.1" (gHue) {channel="hue:LCT001:0017882a573e:8:color"}
```

knx.items:

```
Dimmer      knx_boardroom_dimmer    "Boardroom Dimmer [%s]"          <light>        { knx="0/0/4+<0/0/20, 0/0/5, 0/0/6+0/0/7" } 
```

knx.rules:

```
// Connects KNX boardroom dim % to dim gCloud
rule "KNXLightDimmer"
when
        Item knx_boardroom_dimmer received command
then
        var PercentType percent
        if(knx_boardroom_dimmer.state instanceof PercentType)  {
            percent = knx_boardroom_dimmer.state as PercentType
            sendCommand(gHue, percent)
        }
end
```

bug.sitemap:
sitemap bug label="Bug Sitemap"
{
        Slider item=knx_boardroom_dimmer label="KNX Dimmmer"
        //Slider item=gHue //Server Error 500 prevents Classic UI rendering, works fine on Basic UI
}

The rule still appears to work fine dimming the hue bulbs with the knx dimmer state value. This was handled via the Classic UI which provides INCREASE/DECREASE buttons for the Dimmer.

I'm unable to verify Slider element on the gHue group directly in Classic UI due to [the bug](https://github.com/eclipse/smarthome/issues/1739). In the Basic UI, a Slider element on gHue group works fine without error, I can also verify this is the case with the Slider element for the KNX Dimmer item. It seems specific to something Classic UI is doing?

--- a/bundles/core/org.eclipse.smarthome.core.test/src/test/groovy/org/eclipse/smarthome/core/items/GroupItemTest.groovy
+++ b/bundles/core/org.eclipse.smarthome.core.test/src/test/groovy/org/eclipse/smarthome/core/items/GroupItemTest.groovy
@@ -18,6 +18,7 @@ import org.eclipse.smarthome.core.library.items.NumberItem
 import org.eclipse.smarthome.core.library.items.SwitchItem
 import org.eclipse.smarthome.core.library.types.ArithmeticGroupFunction
 import org.eclipse.smarthome.core.library.types.OnOffType
+import org.eclipse.smarthome.core.library.types.PercentType
 import org.eclipse.smarthome.core.library.types.RawType
 import org.eclipse.smarthome.core.types.RefreshType
 import org.eclipse.smarthome.core.types.UnDefType
@@ -264,5 +265,17 @@ class GroupItemTest extends OSGiTest {
         assertThat events.size(), is(0)
 
         assertTrue groupItem.getState().equals(oldGroupState)
+    }
+
+    @Test
+    void 'assert that group item without function can have a convertible state' (){
+        GroupItem groupItem = new GroupItem("root")
+        PercentType pt = new PercentType(50);
+        groupItem.setState(pt);
+
+        def groupStateAsOnOff = groupItem.getStateAs(OnOffType);
+
+        //any value >0 means on, so 50% means the group state should be ON
+        assertTrue OnOffType.ON.equals(groupStateAsOnOff)
     }
 }

--- a/bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/GroupItem.java
+++ b/bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/GroupItem.java
@@ -239,6 +239,11 @@ public class GroupItem extends GenericItem implements StateChangeListener {
      */
     @Override
     public State getStateAs(Class<? extends State> typeClass) {
+        // if a group does not have a function it cannot have a state
+        if (function == null) {
+            return null;
+        }
+
         State newState = function.getStateAs(getAllMembers(), typeClass);
         if (newState == null && baseItem != null) {
             // we use the transformation method from the base item

--- a/bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/GroupItem.java
+++ b/bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/GroupItem.java
@@ -240,11 +240,11 @@ public class GroupItem extends GenericItem implements StateChangeListener {
     @Override
     public State getStateAs(Class<? extends State> typeClass) {
         // if a group does not have a function it cannot have a state
-        if (function == null) {
-            return null;
+        State newState = null;
+        if (function != null) {
+            newState = function.getStateAs(getAllMembers(), typeClass);
         }
 
-        State newState = function.getStateAs(getAllMembers(), typeClass);
         if (newState == null && baseItem != null) {
             // we use the transformation method from the base item
             baseItem.setState(state);

